src/libs/actions/data-lake.ts

```diff
@@ -33,7 +37,13 @@ export const getCockpitActionVariableInfo = (id: string): CockpitActionVariable
 
 export const createCockpitActionVariable = (variable: CockpitActionVariable): void => {
   if (cockpitActionVariableInfo[variable.id]) {
-    throw new Error(`Cockpit action variable with id '${variable.id}' already exists. Update it instead.`)
+    console.log(`Cockpit action variable with id '${variable.id}' already exists. Renaming to ${variable.name}_new.`)
+    showSnackbar({
+      message: `Cockpit action variable with id '${variable.id}' already exists. Renaming to ${variable.name}_new.`,
+    })
+    variable.id = `${variable.id}_new`
+    variable.name = `${variable.name}_new`
+    cockpitActionVariableInfo[variable.id] = variable
```

Comment:
There's no need for the snackbar here.
You can have the snackbar in the component that is calling `createCockpitActionVariable`.
This is a void method with a throw, so there you can put a try/catch block to deal with error/success case, like so:

```
// EditMenu.vue

import { useSnackbar } from '@/composables/snackbar'

const { showSnackbar } = useSnackbar()

try {
  createCockpitActionVariable(variable)
  showSnackbar({ message: 'Parameter successfully updated', variant: 'success' })
} catch(error) {
  showSnackbar({message: `Failed to add Cockpit action variable with id '${variable.id}'. Name already exists. Please rename it.`})
}
```

If necessary to deal with the renaming process, please do it in the `EditMenu.vue` component. You can do that by adding an `code` property to the error (inside the `data-lake.ts` file) and checking for the error in the Vue component, although I think the more correct approach is to ask the user to rename the variable, so it's clear to them that they already have a variable with that name).